executor/src/constant_evaluator/mod.rs

```diff
@@ -57,23 +57,22 @@ fn generate_values<T: FieldElement>(
             let values: Vec<_> = values
                 .iter()
                 .flat_map(|elements| {
-                    assert!(elements.repetitions >= 1);
                     let items = elements
-                        .values
+                        .values()
                         .iter()
                         .map(|v| evaluator.evaluate(v))
                         .collect::<Vec<_>>();
 
                     let original_len = items.len();
-                    if elements.repetitions == 1 {
+                    if !elements.is_repeated() {
                         items
                     } else if original_len == 1 {
-                        vec![items[0]; elements.repetitions as usize]
+                        vec![items[0]; elements.size() as usize]
```

Comment:
aren't these two cases covered by the third one?

Reply:
They are, I guess this implementation is slightly more efficient?

Reply:
But I'll put some faith in the Rust compiler and remove the first two cases.